road/scatter.py

```diff
@@ -0,0 +1,236 @@
+    })
+    diagonal_df = pd.DataFrame({
+        x_field: [df[x_field].min(), df[x_field].max()],
+        y_field: [df[x_field].min(), df[x_field].max()]  # Diagonal line: y = x
+    })
+    equation_text = f"y = {m:.2f}x + {b:.2f}"
+    max_value = max(df[x_field].max(), df[y_field].max())
+    tooltip_config = [{"field": field,
+                       "type": "nominal" if field in nominal_fields else "quantitative",
+                       "title": field} for field in fields]
+
+    vega_lite_config = {
+        "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+        "description": "A scatterplot with a regression line",
+        "data": {"url": csv_file},
+        "transform": [] if include_all_data else [{"filter": f"datum['{classification_col}'] == '{classification_col_types}'"}],
+        "layer": [
+            {
+                "mark": {
+                    "type": "rule",
+                    "color": "grey",
+                    "strokeWidth": 3
+                },
+                "encoding": {
+                    "y": {"datum": 0}
+                }
+            },
+            {
+                "mark": "point",
+                "encoding": {
+                    "x": {"field": x_field, "type": "quantitative", "scale": {"domain": [0, max_value]}},
+                    "y": {"field": y_field, "type": "quantitative", "scale": {"domain": [0, max_value]}},
+                    "tooltip": tooltip_config
+                }
+            },
+            {
+                "mark": {
+                    "type": "text",
+                    "align": "left",
+                    "baseline": "bottom",
+                    "dx": 5,
+                    "dy": -5
+                },
+                "data": {"values": [{"text": equation_text}]},
+                "encoding": {
+                    "x": {"value": 0},
+                    "y": {"value": 0},
+                    "text": {"field": "text", "type": "nominal"},
+                    "color": {"value": "red"}
+                }
+            },
+            {
+                "mark": "line",
+                "data": {"values": diagonal_df.to_dict(orient='records')},
+                "encoding": {
+                    "x": {"field": x_field, "type": "quantitative"},
+                    "y": {"field": y_field, "type": "quantitative"},
+                    "color": {"value": "blue"}
+                }
+            },
+            {
+                "mark": "line",
+                "data": {"values": regression_df.to_dict(orient='records')},
+                "encoding": {
+                    "x": {"field": x_field, "type": "quantitative"},
+                    "y": {"field": y_field, "type": "quantitative"},
+                    "color": {"value": "red"}
+                }
+            },
+        ]
+    }
+
+    base_name = os.path.splitext(csv_file)[0]
+    file_path = f"{sanitize_filename(classification_col_types)}_{name}.vega.json"

```

Comment:
maybe move the filepath to the function arg? though this would be quite complicated since the filename is constructed from multiple variables hmm...

same comment applies to all the other cases too